### PR TITLE
GS Validator

### PR DESCRIPTION
**Pull request by piontec**
This PR adds:
- a module for easily adding simple validation rules
- two simple validators
  - check for presence of `schema.values.json`
  - check for presence and usage of the GS team label

All changes in `examples` dir are only copied over from the 'hello-world` app repo.

### CODE REVIEW

File: app_build_suite/build_steps/giant_swarm_validators/helm.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        return "F0001"
+
+    def validate(self, config: argparse.Namespace) -> bool:
+        return os.path.exists(os.path.join(config.chart_dir, VALUES_SCHEMA_JSON))
+
+
+class HasTeamLabel:
+
+    escaped_label = re.escape(GS_TEAM_LABEL_KEY)
+    _label_regexp = (
```
Comment (MarcelMue) [inline app_build_suite/build_steps/giant_swarm_validators/helm.py:40]: Do we need to have a regex because the file isn't proper YAML or why can't we just parse the file normally?

Comment (MarcelMue) [inline app_build_suite/build_steps/giant_swarm_validators/helm.py:40]: Do we need to have a regex because the file isn't proper YAML or why can't we just parse the file normally?

File: app_build_suite/build_steps/helm.py [hunk 2]
```diff
@@ -730,6 +734,114 @@ def cleanup(
...
+
+    def _load_giant_swarm_validators(self) -> List[GiantSwarmValidator]:
+        gs_validators: List[GiantSwarmValidator] = []
+        current_frame = inspect.currentframe()
+        if current_frame is None:
+            raise ValidationError(self.name, "Can't check current frame and detect current module's path.")
+
+        cur_mod_path = os.path.dirname(os.path.abspath(inspect.getfile(current_frame)))
+        validators_mod_path = os.path.join(cur_mod_path, "giant_swarm_validators")
+        for modname in listdir(validators_mod_path):
```
Comment (MarcelMue) [inline app_build_suite/build_steps/helm.py:817]: Why do we dynamically load the validators here? I feel like having just imports or something simpler would be sufficient. Basically I don't understand the usecase for loading them dynamically like this.

Comment (MarcelMue) [inline app_build_suite/build_steps/helm.py:817]: Why do we dynamically load the validators here? I feel like having just imports or something simpler would be sufficient. Basically I don't understand the usecase for loading them dynamically like this.

File: app_build_suite/build_steps/giant_swarm_validators/helm.py (hunk 1 above)
Comment (piontec) [reply app_build_suite/build_steps/giant_swarm_validators/helm.py:40]: Yeah, it's because it's not a YAML - we would need a go template parser or maybe even runner :|

**Review by rossf7 — APPROVED** [review]
Looks good. 

Not related to this change but can we drop the 3 from `helm3-build-pipeline` in docs? I don't think it's needed anymore.

File: app_build_suite/build_steps/helm.py (hunk 2 above)
Comment (piontec) [reply app_build_suite/build_steps/helm.py:817]: It's something I wanted to do for some time. When I work on a new module, I want it to be "just used", I often forget to register it where it has to appear. But so far I wanted to keep explicit control over the execution order (for all the main modules).
Here I decided to load it dynamically, so it's really easy for other people to add next simple validators. My idea is that we can easily migrate what Marian already did and maybe some more.
TL;DR: no strong reason. I find it easier this way to add those simple validators. And it's easy and fun :P 
But if you think the complication is not worth the benefits, we can rethink that

**Review by MarcelMue — APPROVED** [review]
